File: Course_01/C Language/Matrizes/diagonal_negativa.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int n;
    printf("Qual a ordem da matriz? ");
    scanf("%d", &n);

    int mat[n][n], negCount = 0;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            printf("Elemento [%d,%d]: ", i, j);
            scanf("%d", &mat[i][j]);
        }
    }

    printf("DIAGONAL PRINCIPAL:\n");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i == j) {
                printf("%d  ", mat[i][j]);
            }
        }
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if ((mat[i][j]) < 0) {
                negCount++;
            }
        }
    }
    printf("\n");
    printf("QUANTIDADE DE NEGATIVOS = %d", negCount);

    return 0;
}

File: Course_01/Java Language/Matrizes/diagonal_negativa.java
package Matrizes;

import java.util.Locale;
import java.util.Scanner;

public class diagonal_negativa {

	public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int i, j, m;
		
		System.out.print("Qual a ordem da matriz? ");
		m = sc.nextInt();
		
		int [][] matriz = new int [m][m];
		
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				matriz[i][j] = sc.nextInt();   
			}
		}
		
		int conteNega = 0;
		
		System.out.println("DIAGONAL PRINCIPAL: ");
		
		for (i = 0; i < m; i++) {
			for (j = 0; j < m; j++) {
				if (i == j) {
					System.out.print(matriz[i][j] + " ");
				}
				if (matriz [i][j] < 0) {
					conteNega++;
				}
			}
		}
		System.out.println();
		System.out.println("QUANTIDADE DE NEGATIVOS = " + conteNega);
		
		sc.close();;
	}

}
